src/builtins/status.rs

```diff
@@ -510,6 +486,55 @@ pub fn status(parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) -> B
             streams.out.append(src);
             return Ok(SUCCESS);
         }
+        STATUS_LOCALE => {
+            cfg_if! {
+                if #[cfg(not(feature = "localize-messages"))]
+                {
+                    streams.err.append(L!("fish was built with the `localize-messages` feature disabled. The `status locale` command is unavailable.\n"));
+                    return Err(STATUS_CMD_ERROR);
+                } else {
+                    if args.is_empty() {
+                        streams.err.append(wgettext_fmt!(
+                            BUILTIN_ERR_MISSING_SUBCMD,
+                            STATUS_LOCALE.to_wstr(),
+                        ));
+                        return Err(STATUS_INVALID_ARGS);
+                    }
+                    match args[0].to_string().as_str() {
+                        "get" => {
+                            streams.out.append(crate::wutil::gettext::status_locale_get());
+                            return Ok(SUCCESS);
+                        }
+                        "set" => {
+                            let owned_langs = args[1..]
+                                .iter()
+                                .map(|lang| lang.to_string())
+                                .collect::<Vec<_>>();
+                            let langs = owned_langs
```

Comment:
`owned_langs` can be a slightly confusing name, and ownership is also implied from the code.
I'd rather call both `langs`;
shadowing locals like this is nice because it reduces effective (lexical?) scope.

Reply:
Renaming `owned_langs` to `langs` would not change any behavior. Ownership and lifetime of both values would still be the same, only that the owned one would be shadowed for good. So changing the interface is not necessary but we definitely can if it looks better.